tools/gen_valid_chars.cc

```diff
@@ -0,0 +1,48 @@
+// generate the atlas valid charsets
+
+#include <array>
+#include <fstream>
+
+void dump_array(std::ostream& os, const std::string& name, const std::array<bool, 256>& chars) {
```

Comment:
how is this file useful?

Reply:
That file is used to generate `spectator/valid_chars.inc`, which is not checked in by the upstream. 

The upstream library calls this tool at build time to generate `valid_chars.inc`. I ended up calling it locally to generate the file.